### PR TITLE
binary packets support and multi listeners ports support.

### PR DESCRIPTION
**Pull request by jfdelnero**
First thanks for this straightforward implementation !

I have added the binary mode support and the multi listeners ports support.

### CODE REVIEW

**Comment by Theldus** [comment]
Hello @jfdelnero,
Thank you so much for your contributions.
Support to binary frames was on my todo list, so thank you for implementing it. Multi listeners looks a great idea too. I just have one consideration to do (see review below).

File: src/ws.c [hunk 1]
```diff
@@ -144,7 +170,7 @@ int ws_sendframe(int fd, const char *msg, bool broadcast)
 		pthread_mutex_lock(&mutex);
 			for (i = 0; i < MAX_CLIENTS; i++)
 			{
-				sock = client_socks[i];
+				sock = client_socks[i].client_sock;
```
Comment (Theldus) [inline src/ws.c:173]: Broadcasts (IMHO) should be restricted to clients connected through the same port, not for all clients. A quick loop could find the port for the broadcast sender and then you can check that port in the broadcast loop. =). If you disagree or have another thoughs on that, please let me know.

**Review by Theldus — CHANGES_REQUESTED** [review]
Besides that, everything looks neat =).

File: src/ws.c (hunk 1 above)
Comment (jfdelnero) [reply src/ws.c:173]: Yes agree but in fact i am currently using wsServer in a global broadcast way (all clients receive the same page update/information). Maybe the "broadcast" parameter type should be changed to pass more options and support both broadcast modes (client broadcast, global broadcast) ?

Comment (jfdelnero) [reply src/ws.c:173]: dismiss my previous answer, i have overlooked your answer. Indeed broadcasting between the ports has very little sense ;)
i will change this point.

Comment (jfdelnero) [reply src/ws.c:185]: Changes done and tested !

**Review by Theldus — APPROVED** [review]
Thank you for reviewing this and for your PR too, I really appreciate, =).
Merged.

**Pull request merged**